python/src/opendp/typing.py

```diff
@@ -330,8 +336,14 @@ def infer_homogeneous(value):
                 infer_homogeneous(public_example.values())
             ])
 
-        if public_example is None:
-            return RuntimeType('Option', [UnknownType("Constructed Option from a None variant")])
+        if isinstance(public_example, Measurement): # pragma: no cover
+            return "AnyMeasurementPtr"
+
+        if isinstance(public_example, Transformation): # pragma: no cover
+            return "AnyTransformationPtr"
+
+        if public_example is None: # pragma: no cover
+            raise UnknownTypeException("Constructed Option from a None variant")
```

Comment:
This error could be clearer... It made sense when the code would make an `UnknownType`, but this raises now.

```suggestion
        if public_example is None: # pragma: no cover
            raise UnknownTypeException("Type of Option cannot be inferred from None")
```